lib/docs/scrapers/typescript.rb

```diff
@@ -2,26 +2,37 @@ module Docs
   class Typescript < UrlScraper
     self.name = 'TypeScript'
     self.type = 'simple'
-    self.release = '3.7.4'
-    self.base_url = 'https://www.typescriptlang.org/docs/'
-    self.root_path = 'tutorial.html'
+    self.release = '4.0.5'
+    self.base_url = 'https://www.typescriptlang.org/docs/handbook'
```

Comment:
This might have been the source of your frustrations with updating this scraper. This page doesn't exist, but does respond with a 301 redirect to `https://www.typescriptlang.org/docs/handbook/`, so it appeared to work.

This is why `root_page?` calls didn't work.